test/test-parse-warning.cpp

```diff
@@ -169,6 +169,162 @@ TEST_F(test_error_equals_does_not_distribute_over_or, non_constant) {
     EXPECT_THAT(p.errors, IsEmpty());
   }
 }
+
+TEST_F(test_parse_warning, warn_on_pointless_string_compare) {
+  {
+    padded_string code(u8"toLowerCase() == 'banana'"_sv);
```

Comment:
I don't think we want to check a bare call to a function named `toLowerCase`. We want to check a call to a *method* named `toLowerCase`. So I think this test should say `s.toLowerCase()` instead.